### ISSUE TITLE
pgo 4.2.2: scaledown when targeting a replica that was once set as master does not work

### ISSUE THREAD

**Issue by jose-joye** [CrunchyData/postgres-operator#1339]
hello,

We are testing the following use case:

1. create cluster (single node)
2. scale cluster
3. manual failover
4. scaledown cluster

--> Error no replica found

**Note:**
Scaling-down a replica that never was a primary works. 
However, if a replica was once set as a primary and back again to a replica --> it is not possible to be scaled down

Example:
```
# create cluster
$pgo create cluster jose-test

$pgo -n cfae-bit-cbcd2 show cluster jose-test

cluster : jose-test (crunchy-postgres-ha:centos7-11.7-4.2.2)
        pod : jose-test-7c6bdfcf64-d8k9n (Running) on l820000002409a (2/2) (primary)
        pvc : jose-test
        resources : CPU Limit=1 Memory Limit=1Gi, CPU Request=0.25 Memory Request=256Mi
        storage : Primary=16Gi Replica=16Gi
        deployment : jose-test
        deployment : jose-test-backrest-shared-repo
        service : jose-test - ClusterIP (172.24.252.185)
        labels : autofail=true crunchy-pgha-scope=jose-test crunchy_collect=true pgouser=admin archive-timeout=1800 current-primary=jose-test pg-cluster=jose-test name=jose-test pg-pod-anti-affinity= pgo-backrest=true pgo-version=4.2.2 workflowid=32908747-37ec-4a9f-97b2-377a9df0c78a crunchy-pgbadger=false deployment-name=jose-test

# scale cluster
$pgo -n cfae-bit-cbcd2 scale jose-test
WARNING: Are you sure? (yes/no): y
created Pgreplica jose-test-xbts

$pgo -n cfae-bit-cbcd2 show cluster jose-test

cluster : jose-test (crunchy-postgres-ha:centos7-11.7-4.2.2)
        pod : jose-test-7c6bdfcf64-d8k9n (Running) on l820000002409a (2/2) (primary)
        pvc : jose-test
        pod : jose-test-xbts-f6cc994d7-grphk (Running) on l820000002403a (2/2) (replica)
        pvc : jose-test-xbts
        resources : CPU Limit=1 Memory Limit=1Gi, CPU Request=0.25 Memory Request=256Mi
        storage : Primary=16Gi Replica=16Gi
        deployment : jose-test
        deployment : jose-test-backrest-shared-repo
        deployment : jose-test-xbts
        service : jose-test - ClusterIP (172.24.252.185)
        service : jose-test-replica - ClusterIP (172.24.76.85)
        pgreplica : jose-test-xbts
        labels : crunchy-pgha-scope=jose-test current-primary=jose-test deployment-name=jose-test pgo-backrest=true crunchy-pgbadger=false crunchy_collect=true pgo-version=4.2.2 pgouser=admin autofail=true name=jose-test pg-pod-anti-affinity= workflowid=32908747-37ec-4a9f-97b2-377a9df0c78a archive-timeout=1800 pg-cluster=jose-test

# failover
$pgo -n cfae-bit-cbcd2 failover  jose-test --target jose-test-xbts
WARNING: Are you sure? (yes/no): y
created Pgtask (failover) for cluster jose-test

$pgo -n cfae-bit-cbcd2 show cluster jose-test

cluster : jose-test (crunchy-postgres-ha:centos7-11.7-4.2.2)
        pod : jose-test-7c6bdfcf64-d8k9n (Running) on l820000002409a (2/2) (replica)
        pvc : jose-test
        pod : jose-test-xbts-f6cc994d7-grphk (Running) on l820000002403a (2/2) (primary)
        pvc : jose-test-xbts
        resources : CPU Limit=1 Memory Limit=1Gi, CPU Request=0.25 Memory Request=256Mi
        storage : Primary=16Gi Replica=16Gi
        deployment : jose-test
        deployment : jose-test-backrest-shared-repo
        deployment : jose-test-xbts
        service : jose-test - ClusterIP (172.24.252.185)
        service : jose-test-replica - ClusterIP (172.24.76.85)
        pgreplica : jose-test-xbts
        labels : autofail=true pg-pod-anti-affinity= pgouser=admin crunchy_collect=true deployment-name=jose-test pgo-version=4.2.2 archive-timeout=1800 crunchy-pgbadger=false crunchy-pgha-scope=jose-test current-primary=jose-test-xbts name=jose-test workflowid=32908747-37ec-4a9f-97b2-377a9df0c78a pg-cluster=jose-test pgo-backrest=true

# scale down
$pgo -n cfae-bit-cbcd2 scaledown  jose-test --query

Cluster: jose-test
REPLICA                 STATUS          NODE            REPLICATION LAG
jose-test               running         l820000002409a             0 MB

$ pgo -n cfae-bit-cbcd2 scaledown  jose-test --target jose-test
WARNING: Are you sure? (yes/no): y
Error: no replicas found for this cluster

```

**Comment by jose-joye** [comment]
Is it ok as a workaround to delete the deployment using kubectl?
eg:
```
kubectl delete deployment jose-test
```

**Comment by andrewlecuyer** [comment]
@jose-joye this is a duplicate of #1201, which describes the same behavior in which it is not possible to scale down the original primary following a failover.  A workaround for this issue is also provided: https://github.com/CrunchyData/postgres-operator/issues/1201#issuecomment-574334957.

Also, please note that a fix for this issue with be included in the next release of the Operator (`v4.3`).
